ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -37,9 +38,12 @@ def behavior_spec_from_proto(
         tuple(DimensionProperty(dim) for dim in obs.dimension_properties)
         for obs in agent_info.observations
     ]
+    sensor_types = [SensorType(obs.sensor_type) for obs in agent_info.observations]
     sensor_specs = [
-        SensorSpec(obs_shape, dim_p)
-        for obs_shape, dim_p in zip(observation_shape, dim_props)
+        SensorSpec(obs_shape, dim_p, sensor_type)
```

Comment:
I think you should rearrange this function; the multiple list comprehensions + zip feel really weird (and probably allocates more memory than necessary for the intermediate lists). Either do something like
```python
sensor_specs = []
for obs in agent_info.observations:
  sensor_specs.append(
    SensorSpec(
      tuple(obs.shape),
      tuple(DimensionProperty(dim) for dim in obs.dimension_properties),
      SensorType(obs.sensor_type) 
    )
```
or the equivalent with a single list comprehension.